### PR TITLE
or): prevent CopyContextToFiberContext panic and add comprehensive test coverage

### PR DESCRIPTION
**Pull request by sixcolors**
## 🐛 Bug Fix

Fixes a panic in `CopyContextToFiberContext` when handling non-pointer context structs and adds comprehensive test coverage for the adaptor middleware.

### Bug Fixes
- **Fixed panic in `CopyContextToFiberContext`**: Added proper handling for both pointer and non-pointer context types
- **Added nil pointer checking**: Prevents crashes when nil contexts are passed
- **Improved type validation**: Gracefully handles non-struct context types

### Test Coverage Improvements
- **297+ lines of new comprehensive tests** covering all error paths and edge cases
- **Unit tests for helper functions**: `resolveRemoteAddr`, `isUnixNetwork`
- **Error handling tests**: IO errors, invalid addresses, context conversion failures
- **Unix socket integration test**: Full end-to-end Unix domain socket testing
- **Benchmark tests**: Performance validation for different payload sizes
- **DRY improvements**: Helper functions `createTestRequest`, `executeHandlerTest` for maintainable tests

### Test Functions Added
- `Test_ConvertRequest` - Comprehensive conversion testing with error handling
- `Test_CopyContextToFiberContext` - Context copying with edge cases
- `Test_HTTPMiddleware_ErrorHandling` - Middleware error path testing
- `Test_FiberHandler_IOError` - IO failure scenarios
- `Test_FiberHandler_WithErrorInHandler` - Handler error propagation
- `Test_resolveRemoteAddr` - Address resolution unit tests
- `Test_isUnixNetwork` - Network type validation
- `Test_FiberHandler_ErrorFallback` - Invalid address fallback behavior
- `Test_FiberHandler_WithUnixSocket` - Unix socket context handling
- `TestUnixSocketAdaptor` - Full Unix socket integration test
- Comprehensive benchmarks for performance validation

## 🧪 Testing

All tests pass and provide comprehensive coverage of:
- ✅ Panic prevention in context copying
- ✅ Unix socket address resolution
- ✅ Error handling and fallback mechanisms  
- ✅ Integration testing with real Unix sockets
- ✅ Performance benchmarking

## 📊 Coverage

This PR significantly improves test coverage for the adaptor middleware, ensuring robust handling of:
- Context type validation and copying
- Unix socket network detection and address resolution
- Error scenarios and graceful fallbacks
- Performance characteristics under various conditions

### CODE REVIEW

File: middleware/adaptor/adaptor.go [hunk 1]
```diff
@@ -53,14 +54,27 @@ func ConvertRequest(c fiber.Ctx, forServer bool) (*http.Request, error) {
 }
 
 // CopyContextToFiberContext copies the values of context.Context to a fasthttp.RequestCtx.
+// This function safely handles struct fields without using unsafe operations where possible.
```
Comment (nickajacks1) [inline middleware/adaptor/adaptor.go]: > without using unsafe operations

and 

> This function uses reflection and unsafe pointers

This sounds contradictory, maybe should reword.

Comment (sixcolors) [reply middleware/adaptor/adaptor.go]: Better?

**Comment by sixcolors** [comment]
@gemini-code-assist what is your suggestion egarding the implementation of the request body size limit to further enhance security by preventing potential HTTP request smuggling vulnerabilities?

**Comment by gaby** [comment]
@sixcolors llm making stuff up 😂
